ci/src/main/scala/org/typelevel/sbt/TypelevelCiPlugin.scala

```diff
@@ -29,23 +29,82 @@ object TypelevelCiPlugin extends AutoPlugin {
 
   object autoImport {
     def tlCrossRootProject: CrossRootProject = CrossRootProject()
+
+    lazy val tlCiHeaderCheck =
+      settingKey[Boolean]("Whether to do header check in CI (default: false)")
+    lazy val tlCiScalafmtCheck =
+      settingKey[Boolean]("Whether to do scalafmt check in CI (default: false)")
```

Comment:
@danicheg does this address your usecase from https://github.com/typelevel/sbt-typelevel/issues/247? You will still have to add the scalafmt sbt plugin to your build, but you can enable the CI check with this setting.